File: src/test/java/com/itmo/microservices/demo/order/controller/OrderControllerTests.java
package java.com.itmo.microservices.demo.order.controller;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.itmo.microservices.demo.order.api.controller.OrderController;
import com.itmo.microservices.demo.order.api.service.OrderService;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
import org.springframework.boot.test.mock.mockito.MockBean;
import org.springframework.http.MediaType;
import org.springframework.security.core.userdetails.User;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.test.context.junit4.SpringRunner;
import org.springframework.test.web.servlet.MockMvc;

import java.util.ArrayList;

import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;

@RunWith(SpringRunner.class)
@WebMvcTest(OrderController.class)
public class OrderControllerTests {
	@MockBean
	OrderService orderService;

	ObjectMapper mapper = new ObjectMapper();

	@Autowired
	MockMvc mockMvc;

	@Test
	public void testCreateNewOrder() throws Exception {
		UserDetails user = new User("username", "password", new ArrayList<>());
		mockMvc.perform(get("")
						.contentType(MediaType.APPLICATION_JSON)
						.content(mapper.writeValueAsString(user)))
				.andExpect(status().isOk());
	}
}

File: src/main/kotlin/com/itmo/microservices/demo/orders/api/controller/OrderControllerTests.kt
/*
package com.itmo.microservices.demo.orders.api.controller

import com.itmo.microservices.demo.DemoServiceApplication
import junit.framework.Assert
import org.junit.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.http.MediaType
import org.springframework.mock.web.MockHttpServletResponse
import org.springframework.test.context.ActiveProfiles
import org.springframework.test.web.servlet.MockMvc
import org.springframework.test.web.servlet.request.MockMvcRequestBuilders

@ActiveProfiles("dev")
@SpringBootTest(classes = [DemoServiceApplication::class])
@AutoConfigureMockMvc
class OrderControllerTests {
    @Autowired
    private lateinit var mockMvc: MockMvc

    @Test
    fun addProduct() {
        var mvcResult: MockHttpServletResponse = mockMvc.perform(
                MockMvcRequestBuilders.post("/orders")
                        .contentType(MediaType.APPLICATION_JSON)
                        .content("""
                 {}
                """.trimIndent())
                        .accept(MediaType.ALL)
        )
                .andReturn()
                .response
                .apply(::println)

        Assert.assertEquals(mvcResult.status, 200)
    }
}*/
